core/chains/evm/txmgr/confirmer_benchmark_test.go

```diff
@@ -0,0 +1,58 @@
+package txmgr_test
+
+import (
+	"testing"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/utils/tests"
+	txmgrcommon "github.com/smartcontractkit/chainlink-framework/chains/txmgr"
+	"github.com/smartcontractkit/chainlink-integrations/evm/assets"
+	"github.com/smartcontractkit/chainlink-integrations/evm/client/clienttest"
+	"github.com/smartcontractkit/chainlink-integrations/evm/config/configtest"
+	"github.com/smartcontractkit/chainlink-integrations/evm/config/toml"
+	"github.com/smartcontractkit/chainlink-integrations/evm/testutils"
+	evmtypes "github.com/smartcontractkit/chainlink-integrations/evm/types"
+	"github.com/smartcontractkit/chainlink/v2/core/internal/cltest"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+)
+
+func BenchmarkEthConfirmer(t *testing.B) {
+	db := testutils.NewSqlxDB(t)
+	txStore := cltest.NewTestTxStore(t, db)
+	ethClient := clienttest.NewClientWithDefaultChainID(t)
+	evmcfg := configtest.NewChainScopedConfig(t, func(c *toml.EVMConfig) {
+		c.GasEstimator.PriceMax = assets.GWei(500)
+	})
+
+	blockNum := int64(100)
+	head := evmtypes.Head{
+		Hash:   testutils.NewHash(),
+		Number: blockNum,
+	}
+	head.IsFinalized.Store(true)
+
+	ethKeyStore := cltest.NewKeyStore(t, db).Eth()
+	_, fromAddress := cltest.MustInsertRandomKeyReturningState(t, ethKeyStore)
+	etx1 := mustInsertConfirmedEthTxWithReceipt(t, txStore, fromAddress, 0, blockNum)
+	etx2 := mustInsertUnconfirmedTxWithBroadcastAttempts(t, txStore, 4, fromAddress, 1, blockNum, assets.NewWeiI(1))
```

Comment:
@huangzhen1997 Think we might want to do the same suggestion Jordan made here too and move the insert into the loop then clear the state at the end of the loop to have a consistent db state each run. We can do something similar to the broadcaster benchmark test